app/src/main/java/com/babylon/wallet/android/presentation/dialogs/preauthorization/PreAuthorizationStatusViewModel.kt

```diff
@@ -0,0 +1,207 @@
+package com.babylon.wallet.android.presentation.dialogs.preauthorization
+
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.viewModelScope
+import com.babylon.wallet.android.data.dapp.IncomingRequestRepository
+import com.babylon.wallet.android.data.repository.PreAuthorizationStatusData
+import com.babylon.wallet.android.data.repository.TransactionStatusClient
+import com.babylon.wallet.android.presentation.common.OneOffEvent
+import com.babylon.wallet.android.presentation.common.OneOffEventHandler
+import com.babylon.wallet.android.presentation.common.OneOffEventHandlerImpl
+import com.babylon.wallet.android.presentation.common.StateViewModel
+import com.babylon.wallet.android.presentation.common.UiState
+import com.babylon.wallet.android.utils.AppEvent
+import com.babylon.wallet.android.utils.AppEventBus
+import com.radixdlt.sargon.AddressFormat
+import com.radixdlt.sargon.TransactionIntentHash
+import com.radixdlt.sargon.extensions.formatted
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.flow.filterIsInstance
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.seconds
+
+@HiltViewModel
+class PreAuthorizationStatusViewModel @Inject constructor(
+    private val incomingRequestRepository: IncomingRequestRepository,
+    private val transactionStatusClient: TransactionStatusClient,
+    private val appEventBus: AppEventBus,
+    savedStateHandle: SavedStateHandle
+) : StateViewModel<PreAuthorizationStatusViewModel.State>(),
+    OneOffEventHandler<PreAuthorizationStatusViewModel.Event> by OneOffEventHandlerImpl() {
+
+    private val args = PreAuthorizationStatusDialogArgs(savedStateHandle)
+    private var pollJob: Job? = null
+    private var isRequestHandled = false
+
+    init {
+        viewModelScope.launch {
+            appEventBus.events
+                .filterIsInstance<AppEvent.Status.PreAuthorization>()
+                .filter {
+                    it.requestId == args.event.requestId
+                }.collect { event ->
+                    val status = State.Status.from(event)
+                    _state.update { it.copy(status = status) }
+
+                    if (event is AppEvent.Status.PreAuthorization.Sent) {
+                        pollTransactionStatus(event)
+                        processExpiration(event.remainingTime)
+                    }
+                }
+        }
```

Comment:
maybe to move this in a fun that describes what it does? 
for example `observePreAuthorizationStatusEvents`?